rust/src/scanner/mod.rs

```diff
@@ -206,20 +208,20 @@ pub mod fake {
 /// Allows starting, stopping and managing the results of new scans.
 pub struct Scanner<S: ScannerStack> {
     running: Arc<RwLock<HashMap<String, RunningScanHandle>>>,
-    storage: Arc<S::Storage>,
+    storage: S::Storage,
```

Comment:
Depending of if we plan to clone a storage within a scanner I think it could be beneficial to enforce an Arc instead of letting the user decide.